library/core/src/mem/mod.rs

```diff
@@ -1052,14 +1052,14 @@ pub const fn copy<T: Copy>(x: &T) -> T {
 #[must_use]
 #[stable(feature = "rust1", since = "1.0.0")]
 #[rustc_const_unstable(feature = "const_transmute_copy", issue = "83165")]
-pub const unsafe fn transmute_copy<Src, Dst>(src: &Src) -> Dst {
+pub const unsafe fn transmute_copy<Src: ?Sized, Dst>(src: &Src) -> Dst {
     assert!(
-        size_of::<Src>() >= size_of::<Dst>(),
+        size_of_val(src) >= size_of::<Dst>(),
         "cannot transmute_copy if Dst is larger than Src"
     );
```

Comment:
This is no longer guaranteed to be compiled out. I do not believe we want this assert in this case.

Reply:
Why is this not guaranteed to be compiled out?

Reply:
@tbu- because `size_of_val` for dynamically-sized types requires a runtime check since the type's size is not known at compile-time